paddle/fluid/platform/variant.h

```diff
@@ -42,3 +42,11 @@ limitations under the License. */
 #include <boost/mpl/comparison.hpp>
 #include <boost/mpl/less_equal.hpp>
 #include <boost/variant.hpp>
+
+// some platform-independent defintion
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__)
```

Comment:
__attribute__((unused)) is supported by gcc and clang. Not related to platform.

Maybe macro should change to 

```
#if __MSVC__
...
#else
...
#endif
```